cli/EAP7-924_Coloured_output_for_CLI.adoc

```diff
@@ -0,0 +1,83 @@
+= EAP7-924 Coloured output for CLI
+:author:            Ingo Weiss
+:email:             ingo@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse/
+
+== Overview
+
+The intent of this feature is to provide a better user experience by means of colouring important parts of the CLI workflow, this includes error messages, warning messages, successful outputs and better visual feedback when using batch and flow control mechanisms if/for/while.
+
+== Issue Metadata
+
+=== Issue:
+
+* {issue-base-url}EAP7-924[EAP7-924]
+
+=== Related Issues:
+
+* {issue-base-url}WFCORE-3450
+* {issue-base-url}WFCORE-3577
+
+=== Dev Contacts:
+
+* mailto:ingo@redhat.com[Ingo Weiss]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts:
+
+* TBD
+
+=== Affected Projects or Components:
+
+* WildFly Core CLI
+
+
+== Requirements
+
+=== Hard requirements
+
+* Ability to enable or disable colour output via CLI argument
+* Ability to enable or disable colour output using `jboss-cli.xml`
+* Ability to configure the colours for each allowed type: `warning`, `error`, `success`, `batch`, `required`
+* Mark required attributes in bold and with a different colour from the default
+* Colour the output of DMR output based on its outcome
+* Add colours to CLI prompt
+
+=== Soft requirements
+
+* Have a different prompt for each workflow step
+
+== Test Plan
+
+The following "matrix" should be tested on each WildFly/EAP supported operating system:
+
+
+[width="75%",options="header,footer"]
+|===
+| Terminal colours? | CLI colour
+| YES | ON
+| YES | OFF
+| NO  | ON
+| NO  | OFF
+|===
+
+== Design Notes
+
+The current prototype assumes the following defaults:
+
+[width="75%",options="header,footer"]
+|===
+| Type | Colour | Intensity
+| Error | Red | Bright
+| Warning | Yellow | Normal
+| Success | Green | Normal
+| Required | Magenta | Bright
+| Batch | Green | Bright
```

Comment:
In case of a workflow command, batch and if/for/try, the whole prompt will change to the specified colour. As part of the prototype I added ... to the prompt to give a better visual feedback when the customer is in workflow mode.

Responding to the comment above, the terminal capabilities are handled by AESH Readline already and, on my limited tests, it works without issues.